contracts/contracts/strategies/NativeStaking/CompoundingValidatorManager.sol

```diff
@@ -488,6 +519,11 @@ abstract contract CompoundingValidatorManager is Governable {
             validator[pubKeyHash].state == VALIDATOR_STATE.STAKED,
             "Validator not staked"
         );
+        require(
```

Comment:
I would rather we move this check inside `stakeETH` into the `if (currentState == VALIDATOR_STATE.REGISTERED` block. So we can prevent staking to a new validator if maximum number is reached

Reply:
Good point. It's too late at `verifyValidator`. I'll change